validator_client/src/duties/mod.rs

```diff
@@ -40,8 +40,9 @@ pub enum Error {
 /// This keeps track of all validator keys and required voting slots.
 pub struct DutiesManager<U: BeaconNode> {
     pub duties_map: RwLock<EpochDutiesMap>,
-    /// A list of all public keys known to the validator service.
-    pub pubkeys: Vec<PublicKey>,
+    /// A list of all signer objects known to the validator service.
+    // TODO: Generalise the signers, so that they're not just keypairs
+    pub signers: Arc<Vec<Keypair>>,
```

Comment:
Does this need to be an Arc?

Reply:
I'm not sure, I made it an Arc because you did in `56d33d2e2`.

Reply:
Yep, I think we'll borrow and read them in the producer and attestor threads. Should be Arc :)